clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -4033,9 +4041,15 @@ void Sema::CheckSYCLKernelCall(FunctionDecl *KernelFunc,
 
 // For a wrapped parallel_for, copy attributes from original
 // kernel to wrapped kernel.
-void Sema::copySYCLKernelAttrs(const CXXRecordDecl *KernelObj) {
+void Sema::copySYCLKernelAttrs(KernelCallOperatorVisitor &KernelCallOperator,
+                               const CXXRecordDecl *KernelFuncObj) {
   // Get the operator() function of the wrapper.
-  CXXMethodDecl *OpParens = getOperatorParens(KernelObj);
+  CXXMethodDecl *OpParens = nullptr;
+
+  if (KernelFuncObj->isLambda())
+    OpParens = KernelFuncObj->getLambdaCallOperator();
+
+  OpParens = KernelCallOperator.getCallOperator();
```

Comment:
Otherwise traverse is done even for lambdas.
```suggestion
  CXXMethodDecl *OpParens = nullptr;

  if (KernelFuncObj->isLambda())
    OpParens = KernelFuncObj->getLambdaCallOperator();
  else
    OpParens = KernelCallOperator.getCallOperator();
```